engine/src/main/java/io/zeebe/engine/nwe/container/CallActivityProcessor.java

```diff
@@ -0,0 +1,196 @@
+import io.zeebe.engine.processor.workflow.deployment.model.element.ExecutableCallActivity;
+import io.zeebe.engine.state.deployment.DeployedWorkflow;
+import io.zeebe.protocol.record.intent.WorkflowInstanceIntent;
+import io.zeebe.protocol.record.value.ErrorType;
+import io.zeebe.util.Either;
+import io.zeebe.util.buffer.BufferUtil;
+
+public final class CallActivityProcessor
+    implements BpmnElementContainerProcessor<ExecutableCallActivity> {
+
+  private static final String UNABLE_TO_COMPLETE_FROM_STATE_MESSAGE =
+      "Expected to complete call activity after child completed, but call activity cannot be completed from state '%s'";
+  private static final String UNABLE_TO_TERMINATE_FROM_STATE_MESSAGE =
+      "Expected to terminate call activity after child terminated, but call activity cannot be terminated from state '%s'";
+
+  private final ExpressionProcessor expressionProcessor;
+  private final BpmnStateTransitionBehavior stateTransitionBehavior;
+  private final BpmnStateBehavior stateBehavior;
+  private final BpmnIncidentBehavior incidentBehavior;
+  private final BpmnEventSubscriptionBehavior eventSubscriptionBehavior;
+  private final BpmnVariableMappingBehavior variableMappingBehavior;
+
+  public CallActivityProcessor(final BpmnBehaviors bpmnBehaviors) {
+    expressionProcessor = bpmnBehaviors.expressionBehavior();
+    stateTransitionBehavior = bpmnBehaviors.stateTransitionBehavior();
+    stateBehavior = bpmnBehaviors.stateBehavior();
+    incidentBehavior = bpmnBehaviors.incidentBehavior();
+    eventSubscriptionBehavior = bpmnBehaviors.eventSubscriptionBehavior();
+    variableMappingBehavior = bpmnBehaviors.variableMappingBehavior();
+  }
+
+  @Override
+  public Class<ExecutableCallActivity> getType() {
+    return ExecutableCallActivity.class;
+  }
+
+  @Override
+  public void onActivating(final ExecutableCallActivity element, final BpmnElementContext context) {
+    variableMappingBehavior
+        .applyInputMappings(context, element)
+        .flatMap(ok -> eventSubscriptionBehavior.subscribeToEvents(element, context))
+        .flatMap(ok -> evaluateProcessId(context, element))
+        .flatMap(this::getWorkflowForProcessId)
+        .flatMap(this::checkWorkflowHasNoneStartEvent)
+        .ifRightOrLeft(
+            workflow -> {
+              final var childWorkflowInstanceKey =
+                  stateTransitionBehavior.createChildProcessInstance(workflow, context);
+
+              final var callActivityInstance = stateBehavior.getElementInstance(context);
+              callActivityInstance.setCalledChildInstanceKey(childWorkflowInstanceKey);
+              stateBehavior.updateElementInstance(callActivityInstance);
+
+              final var callActivityInstanceKey = context.getElementInstanceKey();
+              stateBehavior.copyVariables(
+                  callActivityInstanceKey, childWorkflowInstanceKey, workflow);
+
+              stateTransitionBehavior.transitionToActivated(context);
```

Comment:
I would move the writing of the `activated` event before the `activating` of the child workflow instance. Otherwise, we change the order of the events and for me, it feels more intuitive the other way around :)